### PR TITLE
Simplify our Select ViewModels

### PR DESCRIPTION
**Pull request by heytherewill**
## What's this?
This is one of those 🔥 PRs

## Why do we want this?
We already have a generic and pretty good picker; no need those extra views.

This also collaterally fixes a bug! The SelectWorkspaceViewModel was only allowing the selection of workspaces that were eligible for project creation! This is correct for the `EditProjectViewModel`, but not for the other places where it's used, so things are better now!!!

## How is it done?
🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 

## Review hints
Approach this PR commit by commit

## :squid: Permissions
Only myself

### CODE REVIEW

File: Toggl.Core.UI/ViewModels/Settings/SettingsViewModel.cs [hunk 1]
```diff
@@ -422,24 +429,43 @@ private async Task toggleTimeEntriesGrouping()
...
         private async Task selectDurationFormat()
         {
-            var newDurationFormat = await Navigate<SelectDurationFormatViewModel, DurationFormat, DurationFormat>(currentPreferences.DurationFormat);
+            var validDurationFormats = Enum.GetValues(typeof(DurationFormat)).Cast<DurationFormat>();
+            var durationFormats = validDurationFormats.Select(selectOptionFromDurationFormat);
+            var selectedDurationFormatIndex = validDurationFormats
+                .IndexOf(durationFormat => durationFormat == currentPreferences.DurationFormat);
+
+            var newDurationFormat = await View
+                .Select(Resources.FirstDayOfTheWeek, durationFormats, selectedDurationFormatIndex);
```
[the remark this answers is not in the document]
Comment (heytherewill) [reply Toggl.Core.UI/ViewModels/Settings/SettingsViewModel.cs]: Yeah, please create an issue so we can change that, but I agree with you even though I can't think of a better name

**Comment by heytherewill** [comment]
I honestly prefer the new way; Its more consistent to native iOS styles. But yeah, it can be changed fairly simply. Either way, pinging @maritoggl and @amulware for decision making purposes

File: Toggl.Core.UI/ViewModels/Settings/SettingsViewModel.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (arturs105) [reply Toggl.Core.UI/ViewModels/Settings/SettingsViewModel.cs]: Maybe we could rename it to `Map` :trollface: :

File: Toggl.iOS/Extensions/ViewControllerExtensions.cs [hunk 2]
```diff
@@ -121,8 +121,13 @@ internal static IObservable<T> ShowSelectDialog<T>(this UIViewController viewCon
             {
                 var alertController = UIAlertController.Create(title, null, UIAlertControllerStyle.ActionSheet);
 
-                foreach (var option in options)
+                var listOfOptions = options.ToList();
+
+                for (int i = 0; i < listOfOptions.Count; i++)
                 {
+                    var option = listOfOptions[i];
+                    var style = i == initialSelectionIndex ? UIAlertActionStyle.Cancel : UIAlertActionStyle.Default;
```
Comment (arturs105) [inline Toggl.iOS/Extensions/ViewControllerExtensions.cs]: `style` is never used

**Comment by arturs105** [comment]
I also like the switch to iOS action sheets. It makes the app feel more... _native_

**Comment by juanlaube** [comment]
I prefer the action sheet too, but I just realized it might not work well when there are a lot of options, e.g: a user with _a lot_ of workspaces 🤔

**Comment by Odrakir** [comment]
Action sheets are indeed more “native” than the current thing. But I have the feeling that this is not what they are supposed to do. At least I’m pretty sure they are not supposed to appear after you press a cell with a “>” indicator. 

I think the real native thing to do would be to push a new table view. Like, for example the system calendar does. 

IMG_3592

Or even (maybe) an inline picker, although I've only seen that for dates.

**Comment by maritoggl** [comment]
> An action sheet is a specific style of alert that appears in response to a control or action, and presents a set of two or more choices related to the current context. Use an action sheet to let people initiate tasks, or to request confirmation before performing a potentially destructive operation. On smaller screens, an action sheet slides up from the bottom of the screen. On larger screens, an action sheet appears all at once as a popover.
Ricardo is right that such settings selection lists are not what action sheets are meant for. Noticed that we currently also use the action sheet when selecting workspace in report view - this also seems wrong. Let’s do what Ricardo has proposed and proceed towards having new tableviews instead.

**Comment by amulware** [comment]
I believe this was loco work? Please create issues for anything else that needs to be done so we can schedule things properly.

I'm fine with what Ricardo suggests (though I very much like our own custom styling on iOS, let's not lose the polished look of that please - talk to our designers!), and will 100% block action sheets for these things, they just don't look very good and don't fit our app.

**Comment by Odrakir** [comment]
I would very much like to do those table views on top of William's changes. I found some issues while doing dark mode that should be solved here as well.

**Comment by heytherewill** [comment]
@Odrakir Given that sheets are blocked, I think you should work on this very branch to help getting it merged :)

@amulware Should I create an issue for this so Ricardo can work on it as part of 2.4?

**Comment by heytherewill** [comment]
Also, I just noticed that we using action sheets for workspace selection in the Reports view and for something else on the Calendar View; should we change that there too?

ping @amulware

**Comment by amulware** [comment]
Yes to both - Mari already approved the same above. Just make sure to involve designers.

Though, not sure if there's time to do this in 2.4, do check with the RM.

**Comment by heytherewill** [comment]
pinging @Odrakir For the issues above, since he's the iOS guy for this PR

**Comment by Odrakir** [comment]
The problem with what @arturs105 mentioned is the lack of a Navigation Controller in the project creation screen. The code I wrote, if there's no navigation controller, it presents the selector modally instead of pushing it.

The ideal solution for this is to actually embed all those views in navigation controllers (start, edit, project creation...). But since that's a solution that will take some more time to implement, for now I'm presenting an Action Sheet instead, as we used to do in reports.

**Comment by arturs105** [comment]
@Odrakir Do you mean that navigation controller would solve both of those issues? Even the jumping of the edit project view?

**Comment by Odrakir** [comment]
I think so, because in that case it would be the same modal.

I still need to think this through and write some docs, but if "-|" is a modal transition and "->" is pushed:

Timeline -| Edit TE -> Select Project -| Create project -> Select Workspace

**Pull request merged**